Shared/Samples/Augment reality to show hidden infrastructure/AugmentRealityToShowHiddenInfrastructureView.ARSceneView.swift

```diff
@@ -0,0 +1,177 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import ArcGISToolkit
+import SwiftUI
+
+extension AugmentRealityToShowHiddenInfrastructureView {
+    /// A world scale scene view displaying pipe graphics from a given model.
+    struct ARPipesSceneView: View {
+        /// The view model for scene view in the sample.
+        @ObservedObject var model: SceneModel
+        
+        /// A Boolean value indicating whether the shadow graphics are visible.
+        @State private var shadowsAreVisible = true
+        
+        /// A Boolean value indicating whether the leader line graphics are visible.
+        @State private var leadersAreVisible = true
+        
+        var body: some View {
+            WorldScaleSceneView { _ in
+                SceneView(scene: model.scene, graphicsOverlays: [
+                    model.pipeGraphicsOverlay,
+                    model.shadowGraphicsOverlay,
+                    model.leaderGraphicsOverlay
+                ])
+            }
+            .calibrationButtonAlignment(.bottomLeading)
+            .onCalibrationViewVisibilityChanged { isPresented in
+                model.scene.baseSurface.opacity = isPresented ? 0.6 : 0
+            }
+            .toolbar {
```

Comment:
The toolbar layout has the issue from `swift/5112`. Here is a [workaround](https://github.com/Esri/arcgis-maps-sdk-swift-samples/pull/353/commits/0ec4b21f7f5ea71714a166a866611c455a436d67).

Reply:
@des12437 I tried using the following but still got the issue. Am I missing something?

```swift
var body: some View {
    if #available(iOS 16, *) {
        NavigationStack {
            worldScaleSceneView
        }
    } else {
        NavigationView {
            worldScaleSceneView
        }
    }
}

@MainActor @ViewBuilder private var worldScaleSceneView: some View {
    VStack(spacing: 0) {
        WorldScaleSceneView { _ in
            SceneView(scene: model.scene, graphicsOverlays: [
                model.pipeGraphicsOverlay,
                model.shadowGraphicsOverlay,
                model.leaderGraphicsOverlay
            ])
        }
        .calibrationButtonAlignment(.bottomLeading)
        .onCalibratingChanged { newCalibrating in
            model.scene.baseSurface.opacity = newCalibrating ? 0.6 : 0
        }
    }
    .toolbar {
        ToolbarItem(placement: .bottomBar) {
            settingsMenu
        }
    }
}
```

Reply:
Yes, add `Divider()` at the bottom of the `VStack`.

Reply:
The code should look like this, the current implementation works due to the `Divider()` but also adds an unnecessary separator line to the toolbar.
```
var body: some View {
    if #available(iOS 16, *) {
        NavigationStack {
            worldScaleSceneView
        }
    } else {
        NavigationView {
            worldScaleSceneView
        }
    }
}

@MainActor @ViewBuilder private var worldScaleSceneView: some View {
    VStack(spacing: 0) {
        WorldScaleSceneView { _ in
            SceneView(scene: model.scene, graphicsOverlays: [
                model.pipeGraphicsOverlay,
                model.shadowGraphicsOverlay,
                model.leaderGraphicsOverlay
            ])
        }
        .calibrationButtonAlignment(.bottomLeading)
        .onCalibratingChanged { newCalibrating in
            model.scene.baseSurface.opacity = newCalibrating ? 0.6 : 0
        }
        Divider()
    }
    .toolbar {
        ToolbarItem(placement: .bottomBar) {
            settingsMenu
        }
    }
}
```

Reply:
@des12437 gotcha. It looks like I was able to remove it without the `NaivgationView` or `@ViewBuilder`. Were those needed for something else?

Reply:
No I tried your branch and it looks like using the `NaivgationView` or `@ViewBuilder` is not necessary. Also the screenshots should be updated since a toolbar is now used.